cmd/torrent-verify/main.go

```diff
@@ -16,16 +16,40 @@ import (
 )
 
 var (
-	filePath = flag.String("torrent", "/path/to/the.torrent", "path of the torrent file")
-	dirPath  = flag.String("path", "/torrent/data", "path of the torrent data")
+	torrentPath = flag.String("torrent", "/path/to/the.torrent", "path of the torrent file")
+	dataPath    = flag.String("path", "/torrent/data", "path of the torrent data")
+	summary     = flag.Bool("summary", false, "display summary at the end")
 )
 
-func init() {
-	flag.Parse()
+func verifySummary(sMap map[bool][]int) {
+	fmt.Println("----------------")
+	fmt.Println(" TORRENT-VERIFY ")
+	fmt.Println("----------------")
+	fmt.Printf("Number of correct pieces: %d\n", len(sMap[true]))
+	fmt.Printf("Number of wrong pieces: %d\n", len(sMap[false]))
+}
+
+func fileToMmap(filename string, length int64, devZero *os.File, mMapSpan *mmap_span.MMapSpan) {
+	osFile, err := os.Open(filename)
+	if err != nil {
+		log.Fatal(err)
+	}
+	mmapFd := osFile.Fd()
+	goMMap, err := gommap.MapRegion(mmapFd, 0, length, gommap.PROT_READ, gommap.MAP_PRIVATE)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if int64(len(goMMap)) != length {
+		log.Printf("file mmap has wrong size: %#v", filename)
+	}
+	osFile.Close()
+	mMapSpan.Append(goMMap)
```

Comment:
Yep, great suggestion. That way the function name would also be more correct semantically :+1: